pallets/rps/src/lib.rs

```diff
@@ -14,89 +13,427 @@ mod tests;
+		StorageMap<_, Blake2_128Concat, ChallengeId, ChallengeState<T::AccountId, BalanceOf<T>>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn challenge_plays_store)]
+	pub type ChallengePlaysStore<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		ChallengeId,
+		Blake2_128Concat,
+		T::AccountId,
+		ChallengePlayHash,
+	>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Triggered when a new challenge has been created. [challenge_id, creator_id, bet_amount]
+		ChallengeCreated(ChallengeId, T::AccountId, BalanceOf<T>),
+		/// Triggered when an account accepts a challenge. [challenge_id, challenger_id]
+		EnteredChallenge(ChallengeId, T::AccountId),
+		/// Triggered when an account plays in a certain challenge. [challenge_id, challenger_id]
+		PlayedInChallenge(ChallengeId, T::AccountId),
+		/// Triggered when both players have sent their play on a given challenge. [challenge_id]
+		ChallengeReadyForReveal(ChallengeId),
+		/// Triggered when a challenge has been finished. [winner_id]
+		ChallengeFinished(Option<T::AccountId>),
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The bet amount is not inferior to the minimum bet value
+		InsufficientBetAmount,
+		/// The challenge identifier could not be located
+		ChallengeNotFound,
+		/// The challenge is not open, so it cannot be entered
+		ChallengeNotOpen,
+		/// Cannot accept challenges created by the same account
+		CannotChallengeOneself,
+		/// Cannot play in a challenge in which the account is not a participant
+		CannotPlayInNonParticipatingChallenge,
+		/// Cannot play in a challenge not in the 'Accepted' state
+		ChallengeStateForbidsPlay,
+		/// Cannot play again in an already played challenge
+		ChallengeAlreadyPlayed,
+		/// Challenge state is not Accepted so it cannot be resolved
+		ChallengeStateForbidsResolution,
+		/// The challenge logic reached an invalid state
+		InvalidState,
+		/// Cannot reveal a challenge in which the account is not a participant
+		CannotRevealNonParticipatingChallenge,
+		/// The hash of the original account play and the value indicated in the reveal don't match
+		InvalidHandHash,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000)]
+		pub fn create_challenge(origin: OriginFor<T>, bet_amount: BalanceOf<T>) -> DispatchResult {
+			let challenger = ensure_signed(origin)?;
+
+			let min_amount = T::MinBetAmount::get();
+			ensure!(bet_amount >= min_amount, Error::<T>::InsufficientBetAmount);
+
+			let challenge_id = NextBetId::<T>::get();
+			let challenge_state =
+				ChallengeState::Open(OpenChallenge { challenger: challenger.clone(), bet_amount });
+
+			NextBetId::<T>::mutate(|x| *x += 1);
+
+			ChallengeStore::<T>::insert(&challenge_id, challenge_state);
+
+			Self::deposit_event(Event::ChallengeCreated(challenge_id, challenger, bet_amount));
+
+			Ok(().into())
```

Comment:
This one can just be:
```suggestion
			Ok(())
```

[DispatchResult is just Result<(), sp_runtime::DispatchError>](https://github.com/paritytech/substrate/blob/25417f5d3c2cf191771f33b738a96eebdcde23a7/frame/support/src/dispatch.rs#L52)